core/ecschema-metadata/src/UnitQuery/UnitQuery.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
+import { SchemaContext, SchemaItem, SchemaItemKey, SchemaKey, Unit } from "../ecschema-metadata";
+import { SchemaItemType } from "../ECObjects";
+import { UNIT_DATA } from "./UnitsData";
+
+export class UnitQuery {
+  /**
+   * @param _context
+   */
+  constructor(private readonly _context: SchemaContext) {}
+
+  /**
+   * Find unit in a schema that has unitName
+   * @param unitName Full name of unit
+   * @returns Unit whose full name matches unitName
+   */
+  public async findUnitByName(unitName: string): Promise<Unit> {
```

Comment:
Not quite sure why this method is needed, the method `SchemaContext.getSchemaItem<Unit>` should handle this exact case. It seems like it's being re-written here.

Reply:
This method uses `SchemaContext.getSchemaItem`, but first creates the SchemaItemKey it needs from a string, so it is like a wrapper method.